### PR TITLE
chore(java-shell): update MongoDB driver to 4.7.1

### PR DESCRIPTION
**Pull request by kornilova203**

### CODE REVIEW

**Review by addaleax — APPROVED** [review]

File: packages/java-shell/src/test/resources/db/createCollectionWithTimeseries.js [hunk 1]
```diff
@@ -7,18 +7,18 @@ db.createCollection('coll', {
...
     {
         "metadata": { "sensorId": 5578, "type": "temperature" },
-        "timestamp": ISODate("2021-05-18T00:00:00.000Z"),
+        "timestamp": new ISODate("2022-09-09T00:00:00.000Z"),
         "temp": 12
     });
 // command
 db.coll.findOne({
-    "timestamp": ISODate("2021-05-18T00:00:00.000Z")
+    "timestamp": new ISODate("2022-09-09T00:00:00.000Z")
```
Comment (addaleax) [inline packages/java-shell/src/test/resources/db/createCollectionWithTimeseries.js:21]: Are these changes necessary? They seem okay but shouldn’t make a difference since `ISODate` is not a constructor

Comment (kornilova203) [reply packages/java-shell/src/test/resources/db/createCollectionWithTimeseries.js:21]: Not necessary but just looks better to me

**Comment by addaleax** [comment]
Fyi, I merged the base branch here because the base of PR was a version from May, so CI was failing due to outdated CI configs.

**Comment by addaleax** [comment]
@kornilova203 Getting some failures with MongoDB 4.0.x in CI:

```
 [2022/09/12 12:36:48.899] com.mongodb.mongosh.DbTest > testCreateCollectionWithTimeseries FAILED
 [2022/09/12 12:36:48.900]     org.junit.ComparisonFailure: expected:<...expireAfterSeconds: [86400, timeseries: { timeField: "timestamp", metaField: "metadata", granularity: "minutes" } }' on server %mongohostport%. The full response is {"ok": 0.0, "errmsg": "The field 'expireAfterSeconds' is not a valid collection option. Options: { capped: false, expireAfterSeconds: 864]00, timeseries: { ti...> but was:<...expireAfterSeconds: [315360000, timeseries: { timeField: "timestamp", metaField: "metadata", granularity: "minutes" } }' on server %mongohostport%. The full response is {"ok": 0.0, "errmsg": "The field 'expireAfterSeconds' is not a valid collection option. Options: { capped: false, expireAfterSeconds: 3153600]00, timeseries: { ti...>
 [2022/09/12 12:36:48.900]         at org.junit.Assert.assertEquals(Assert.java:115)
 [2022/09/12 12:36:48.900]         at org.junit.Assert.assertEquals(Assert.java:144)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.TestUtilKt.compare(testUtil.kt:190)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.TestUtilKt.access$compare(testUtil.kt:1)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.TestUtilKt$doTest$4.invoke(testUtil.kt:98)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.TestUtilKt$doTest$4.invoke(testUtil.kt:71)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.TestUtilKt.withDb(testUtil.kt:168)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.TestUtilKt.doTest(testUtil.kt:71)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.TestUtilKt.doTest$default(testUtil.kt:36)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.DbTest$test$1.invoke(DbTest.kt:29)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.DbTest$test$1.invoke(DbTest.kt:28)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.ShellTestCase.withShell(ShellTestCase.kt:29)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.DbTest.test(DbTest.kt:28)
 [2022/09/12 12:36:48.900]         at com.mongodb.mongosh.DbTest.testCreateCollectionWithTimeseries(DbTest.kt:8)
```

**Comment by kornilova203** [comment]
@addaleax, thanks! I've pushed a fix
